### ISSUE TITLE
Turning off accessibility leaves markers on charts even with markers enabled false

### ISSUE THREAD

**Issue by XiLiuRoy** [highcharts/highcharts#16624]
#### Expected behaviour
When markers are not enabled for a chart, turning on or off Accessibility flag should not leave any markers on the chart after hovering.

#### Actual behaviour
A point marker left on the chart when hover over different point or move mouse away from widget.

#### Live demo with steps to reproduce

https://jsfiddle.net/qsno4kxu/2/ -> version 9.3.1
https://jsfiddle.net/24mx7h6w/14/ -> version 9.1.2

1. Open the above jsfiddle example
2. Marker is disabled in series config
2. Click "Disable A11Y" button
3. Use mouse to hover through the chart
4. Markers are left on the chart after hovering

#### Product version
Highcharts 9.3.1
Highcharts 9.1.2

#### Affected browser(s)
Chrome, Firefox, IE

**Comment by raf18seb** [comment]
Hi @XiLiuRoy, thanks for creating the ticket!

`Internal note:`
Working fine in 8.1.2, broken since 8.2.0. Unable to bisect.

Probably related to https://github.com/highcharts/highcharts/issues/15480

**Comment by oysteinmoseng** [comment]
Internal note: Only happens with category or datetime x-axes for some bizarre reason.
